### PR TITLE
add support for redis v9

### PR DESCRIPTION
**Pull request by YafimK**
This updates the support for go-redis libary to latest v9

### CODE REVIEW

**Review by Shivam010 — CHANGES_REQUESTED** [review]
The changes look good to me. Thanks @YafimK for the PR.

Additionally, it seems that with go-redis@v9 we can support Go versions: [ 1.13, 1.14, 1.15 ], which is causing tests to fail. **Can you update the go version in the workflow file to [ 1.20 ] ?**
https://github.com/nitishm/go-rejson/blob/master/.github/workflows/test.yml#L9

Also, the linters in code-analysis.yml are also failing due to some version issues. Are we willing to look into it? We can have a separate PR also for the linters fix.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go_version: [ 1.13, 1.14, 1.15 ]
+        go_version: [ 1.13, 1.14, 1.15, 1.20 ]
```
Comment (Shivam010) [inline .github/workflows/test.yml:9]:
```suggestion
        go_version: [ 1.20 ]
```

**Comment by d0x7** [comment]
Seems all requested changes were made to me, right? When is this due to merge? I'd be more than happy to be able to use this.

**Review by Shivam010 — APPROVED** [review]

**Comment by Shivam010** [comment]
Sorry for the delay. Looks like we are all set. Thanks for the PR @YafimK

**Pull request merged**